src/authorization-response/Payload.ts

```diff
@@ -19,11 +19,13 @@ export const createResponsePayload = async (
 
   // If state was in request, it must be in response
   const state: string | undefined = await authorizationRequest.getMergedProperty('state');
+  const payload = await authorizationRequest.requestObject?.getPayload();
 
   const responsePayload: AuthorizationResponsePayload = {
     ...(responseOpts.accessToken && { access_token: responseOpts.accessToken }),
     ...(responseOpts.tokenType && { token_type: responseOpts.tokenType }),
     ...(responseOpts.refreshToken && { refresh_token: responseOpts.refreshToken }),
+    ...(payload?.nonce && { nonce: payload.nonce }),
```

Comment:
Nonce should not endup in authorization response. That should be either in the id token or vp token

Reply:
```
const oid4vp = await verifyPresentations(this, verifyOpts);
const nonce = merged.nonce ?? verifiedIdToken?.payload.nonce ?? oid4vp.nonce;
```
When I remove nonce from AuthorizationResponsePayload  it will not be in merged.nonce, I don't have idToken so I will need to debug verifyPresentations .

Reply:
As mentioned it needs to end either in the id token and/or vp token depending om which token(s) are requested. Since this comes from the OID request object it should never end up in oauth2 response payload directly

Reply:
So if there is no request object then there indeed is no nonce typically

Reply:
I reverted this. After retest with newer libraries this appears to be working again.